packages/app/src/components/SourceForm.tsx

```diff
@@ -536,6 +537,243 @@ export function TraceTableModelForm({
+        >
+          <SQLInlineEditorControlled
+            connectionId={connectionId}
+            database={databaseName}
+            table={tableName}
+            control={control}
+            name="serviceNameExpression"
+            placeholder="ServiceName"
+          />
+        </FormRow>
+        <FormRow
+          label={'Resource Attributes Column'}
+          helpText="Column containing resource attributes/tags associated with the metric"
+        >
+          <SQLInlineEditorControlled
+            connectionId={connectionId}
+            database={databaseName}
+            table={tableName}
+            control={control}
+            name="resourceAttributesExpression"
+            placeholder="ResourceAttributes"
+          />
+        </FormRow>
+        <FormRow
+          label={'Metric Unit Column'}
+          helpText="Column containing the unit of measurement for the metric"
+        >
+          <SQLInlineEditorControlled
+            connectionId={connectionId}
+            database={databaseName}
+            table={tableName}
+            control={control}
+            name="metricUnitExpression"
+            placeholder="MetricUnit"
+          />
+        </FormRow>
+        <FormRow
+          label={'Metric Flag Column'}
+          helpText="Column containing flags or markers associated with the metric"
+        >
+          <SQLInlineEditorControlled
+            connectionId={connectionId}
+            database={databaseName}
+            table={tableName}
+            control={control}
+            name="flagsExpression"
+            placeholder="Flags"
+          />
+        </FormRow>
+        <FormRow
+          label={'Event Attributes Expression'}
+          helpText="Column containing additional attributes/dimensions for the metric"
+        >
+          <SQLInlineEditorControlled
+            connectionId={connectionId}
+            database={databaseName}
+            table={tableName}
+            control={control}
+            name="eventAttributesExpression"
+            placeholder="Attributes"
+          />
+        </FormRow>
+      </Stack>
+    </>
+  );
+}
+
+function TableModelForm({
+  control,
+  watch,
+  setValue,
+  kind,
+}: {
+  control: Control<TSource>;
+  watch: UseFormWatch<TSource>;
+  setValue: UseFormSetValue<TSource>;
+  kind: SourceKind;
+}) {
+  switch (kind) {
+    case SourceKind.Log:
+      return (
+        <LogTableModelForm
+          control={control}
+          watch={watch}
+          setValue={setValue}
+        />
+      );
+    case SourceKind.Trace:
+      return (
+        <TraceTableModelForm
+          control={control}
+          watch={watch}
+          setValue={setValue}
+        />
+      );
+    case SourceKind.Metric:
+      return (
+        <MetricTableModelForm
+          control={control}
+          watch={watch}
+          setValue={setValue}
+        />
+      );
+    case SourceKind.Session:
```

Comment:
The session reuses the same clickhouse log exporter, which means it should be able to share `LogTableModelForm` component

Reply:
Fixed.